### PR TITLE
JUCX: fix some hangs in UcpListenerTest.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
1. Iterate over network interfaces in reverse order to first try to bind listener on RDMA capable network interface.
2. For all endpoints use peerErrorHandling mode
3. Progress stream send/recv on correct workers

## Why ?
To fix JUCX tests hangs like here: https://dev.azure.com/ucfconsort/0b36e3f0-8ab9-4a48-b68b-4b2350e02c88/_apis/build/builds/15292/logs/219

*It does not fix the fundamental problem (when using non-RDMA NIC sometimes connection handler comes with wrong sockaddr), but at least will not break tests. On RDMA capable NICs didn't reproduce

### CODE REVIEW

**Comment by yosefe** [comment]
> *It does not fix the fundamental problem (when using non-RDMA NIC sometimes connection handler comes with wrong sockaddr), but at least will not break tests. On RDMA capable NICs didn't reproduce

Do you think it's an issue with TCP connection manager?

**Comment by petro-rudenko** [comment]
The issue happens on creating backward connection (from server to client) by connection handler:

```
#3  0x00007f49d1c7fad4 in ucs_handle_error (message=0x7f4a858ab540 "Fatal: client sockaddr data contains invalid address mode 0") at d
ebug/debug.c:1078
#4  0x00007f49d1c7c2e2 in ucs_fatal_error_message (file=0x7f49d0723cf6 "core/ucp_ep.c", line=639, function=0x7f49d07251b0 <__FUNCTION_
_.15619> "ucp_ep_create_server_accept", message_buf=0x7f4a858ab540 "Fatal: client sockaddr data contains invalid address mode 0") at d
ebug/assert.c:37
#5  0x00007f49d1c7c462 in ucs_fatal_error_format (file=0x7f49d0723cf6 "core/ucp_ep.c", line=639, function=0x7f49d07251b0 <__FUNCTION__
.15619> "ucp_ep_create_server_accept", format=0x7f49d0724270 "Fatal: client sockaddr data contains invalid address mode %d") at debug/
assert.c:53
#6  0x00007f49d060875b in ucp_ep_create_server_accept (worker=0x7f4a7e3af050, conn_request=0x7f4974001630, ep_p=0x7f4a858abb28) at cor
e/ucp_ep.c:638
#7  0x00007f49d0608866 in ucp_ep_create_api_conn_request (worker=0x7f4a7e3af050, params=0x7f4a858abbf0, ep_p=0x7f4a858abb70) at core/u
cp_ep.c:671
#8  0x00007f49d0608ef0 in ucp_ep_create (worker=0x7f4a7e3af050, params=0x7f4a858abbf0, ep_p=0x7f4a858abcc8) at core/ucp_ep.c:803
#9  0x00007f49d03c4f08 in Java_org_openucx_jucx_ucp_UcpEndpoint_createEndpointNative (env=0x7f4a7c0099f8, jucx_ep=0x7f4a858abdc8, ucp_
ep_params=0x7f4a858abdc0, worker_ptr=139957922099280) at endpoint.cc:94

>>> p *conn_request
$22 = {
  listener = 0x7f4a7e216050,
  uct_listener = 0x0,
  uct_req = 0x18000000001,
  cm_idx = 255 '\377',
  dev_name = "\377\377\377\377\377\377\377\025\000\000\000\000\000\000\000\377\377\377\377", '\000' <repeats 12 times>,
  remote_dev_addr = 0x0,
  client_address = {
    ss_family = 65535,
    __ss_padding = "\377\377", '\000' <repeats 52 times>, "\250\026\000tI\177\000\000\000\000\000\000\000\000\000\000\270\026\000tI\17
7\000\000\230ݛ\177J\177\000\000\330e!~J\177\000\000\061\371\001", '\000' <repeats 20 times>,
    __ss_align = 0
  },
  ep = 0x0,
  sa_data = {
    ep_id = 0,
    err_mode = 0 '\000',
    addr_mode = 0 '\000',
    dev_index = 0 '\000'
  }
}
```

**Comment by yosefe** [comment]
@alinask can you pls take a look?

File: bindings/java/src/test/java/org/openucx/jucx/UcpListenerTest.java [hunk 1]
```diff
@@ -86,7 +87,7 @@ public void testConnectionHandler() throws Exception {
         UcpListener clientListener = tryBindListener(clientWorker, listenerParams);
 
         UcpEndpoint clientToServer = clientWorker.newEndpoint(new UcpEndpointParams()
-            .setSocketAddress(serverListener.getAddress()));
+            .setPeerErrorHandlingMode().setSocketAddress(serverListener.getAddress()));
```
Comment (yosefe) [inline bindings/java/src/test/java/org/openucx/jucx/UcpListenerTest.java:90]: should we setErrorHandler() ?

**Review by yosefe — APPROVED** [review]

**Comment by petro-rudenko** [comment]
bot:pipe:retest

**Pull request merged**